sway/input/seat.c

```diff
@@ -211,42 +270,74 @@ static void handle_focus_destroy(struct wl_listener *listener, void *data) {
 }
 
 void sway_seat_set_focus(struct sway_seat *seat, swayc_t *container) {
-	swayc_t *last_focus = seat->focus;
+	swayc_t *last_focus =
+		(seat->has_focus ? sway_seat_get_focus(seat, &root_container) : NULL);
```

Comment:
Yeah seeing all of this boilerplate everywhere I am super on board with `sway_seat_get_focus` returning NULL and ditching `has_focus`